integrations/crewai/src/ucai_crewai/toolkit.py

```diff
@@ -0,0 +1,170 @@
+import json
+import logging
+from typing import Any, Callable, Dict, List, Optional
+
+from crewai_tools import BaseTool as CrewAIBaseTool
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from ucai.core.client import BaseFunctionClient
+from ucai.core.utils.client_utils import validate_or_set_default_client
+from ucai.core.utils.function_processing_utils import (
+    generate_function_input_params_schema,
+    get_tool_name,
+    process_function_names,
+)
+
+_logger = logging.getLogger(__name__)
+
+# NB: some CrewAI arguments cannot be inferred from the UC payload and instead must come from the 
+# user. We dynamically pull these arguments from kwargs based on keyword name.
+_CREWAI_KWARGS_FROM_USER = ["description_updated", "cache_function", "result_as_answer"]
```

Comment:
Can these be set during tool creation?

Reply:
(for instance, LlamaIndex has a similar concept to what I imagine the funtionality of `result_as_answer` option is - exposed in the constructor as `return_direct`. If these configurations impact the way that an agent will use a tool, it will probably be best to have them be explicitly declared through the tool definition interface.

Reply:
Can we also add a link where these arguments come from?

Reply:
Refactoring but adding a link in the class